### PR TITLE
Add CppAD

### PR DESCRIPTION
**Pull request by traversaro**
Updated version of https://github.com/conda-forge/staged-recipes/pull/11300 .

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
@jcarpent I added you as a maintainer as you were the original author of https://github.com/conda-forge/staged-recipes/pull/11300, if you prefer to be removed feel free to let me know, thanks!

**Comment by jcarpent** [comment]
This is fine for me. Thanks @traversaro.

**Comment by traversaro** [comment]
The PR is now ready to review: 
@conda-forge/help-c-cpp As the library is written in C++ 
@wolfv as the library is used in robotics (it is a dependency of https://github.com/stack-of-tasks/pinocchio, for example)

**Comment by jcarpent** [comment]
Great job @traversaro. Thanks for handling it.

**Comment by jcarpent** [comment]
By the way, the Python bindings for CppAD in Pinocchio are coming soon.

File: recipes/cppad/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - make  # [linux]
+   
+test:
+  commands:
```
Comment (xhochy) [inline recipes/cppad/meta.yaml:28]: Can you test for the existence of the shared library here or is this header-only?

Comment (traversaro) [reply recipes/cppad/meta.yaml:28]: I added the tests in https://github.com/conda-forge/staged-recipes/pull/13628/commits/5e92cf08723cbea0f6a936367274666d586f26cd . For now on Windows the static library is compiled as the use of static libraries is supported in the upstream project, but I plan to eventually patch the upstream project and import the patch to generate shared libraries also on Windows, for [CFEP-18 compliance](https://github.com/conda-forge/cfep/blob/master/cfep-18.md).

**Comment by wolfv** [comment]
this might be interesting for @tkralphs as well.

**Pull request merged**